jscomp/bsb/bsb_main.ml

```diff
@@ -354,36 +357,40 @@ let () =
         ->
         begin
           Arg.parse bsb_main_flags handle_anonymous_arg usage;
-          (* [-make-world] should never be combined with [-package-specs] *)
-          let make_world = !make_world in 
-          begin match make_world, !force_regenerate with
-            | false, false -> 
-              if !watch_mode then begin
-                watch_exit ()
-                (* ninja is not triggered in this case
-                   There are several cases we wish ninja will not be triggered.
-                   [bsb -clean-world]
-                   [bsb -regen ]
-                *)
-              end 
-            | make_world, force_regenerate ->
-              (* don't regenerate files when we only run [bsb -clean-world] *)
-              let config_opt = regenerate_ninja ~generate_watch_metadata:true ~override_package_specs:None ~no_dev:false ~forced:force_regenerate cwd bsc_dir  in
-              if make_world then begin
-                make_world_deps config_opt
-              end;
-              if !watch_mode then begin
-                watch_exit ()
-                (* ninja is not triggered in this case
-                   There are several cases we wish ninja will not be triggered.
-                   [bsb -clean-world]
-                   [bsb -regen ]
-                *)
-              end else if make_world then begin
-                ninja_command_exit  vendor_ninja [||] 
-              end
-          end;
 
+          (* first, check whether we're in boilerplate generation mode, aka -init foo -theme bar *)
+          match !generate_theme_with_path with
```

Comment:
Indentation changed but this switch is basically the only relevant change. I didn't duplicate the logic into the case where we're using `bsb bla -- ninjaOptionHere`, because that doesn't make sense